dask_match/core.py

```diff
@@ -100,22 +100,22 @@ def __repr__(self):
     def __getattr__(self, key):
         if key == "__name__":
             return object.__getattribute__(self, key)
-        elif key in type(self)._parameters:
-            idx = type(self)._parameters.index(key)
-            return self.operands[idx]
         elif key in dir(type(self)):
             return object.__getattribute__(self, key)
         elif is_dataframe_like(self._meta) and key in self._meta.columns:
             return self[key]
         else:
             return object.__getattribute__(self, key)
 
+    def operand(self, key):
+        return self.operands[type(self)._parameters.index(key)]
```

Comment:
Note that commit [6f191e4](https://github.com/mrocklin/dask-match/pull/2/commits/6f191e46381f37d5c1b0c9ee20bc581cce4c46c1) demonstrates one possible approach to the proposal in #4 (adding a distinct `operand` API for accessing parameters used to create the expression). Although this prohibits us from using `self.<operand>` syntax, I feel that sacrificing this short-hand makes it much easier to avoid recursion traps and unexpected attribute/column-name collisions.